src/pages/collators/dialogs/ClaimRewardsDialog.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useMemo, useState } from "preact/hooks";
+import { nativeToDecimal, format } from "../../../helpers/parseNumbers";
+import { Button, Modal } from "react-daisyui";
+import LabelledInputField from "../../../components/LabelledInputField";
+import { h } from "preact";
+import SuccessDialogIcon from "../../../assets/success-dialog";
+import { CloseButton } from "../../../components/CloseButton";
+import { useGlobalState } from "../../../GlobalStateProvider";
+import { useNodeInfoState } from "../../../NodeInfoProvider";
+import { getErrors } from "../../../helpers/substrate";
+import { toast } from "react-toastify";
+
+interface Props {
+  userRewardsBalance?: string;
+  inflationInfo?: ParachainStakingInflationInflationInfo;
+  tokenSymbol?: string;
+  visible?: boolean;
+  onClose: () => void;
+  onSubmit?: (amount: string) => void;
+}
+
+enum ClaimStep {
+  Start = 0,
+  Confirm = 1,
+  Success = 2
+}
+
+function ClaimRewardsDialog(props: Props) {
+  const {
+    userRewardsBalance = "0",
+    tokenSymbol,
+    visible,
+    onClose,
+  } = props;
+
+  const { createClaimRewardExtrinsic } = useStakingPallet();
+  const { api } = useNodeInfoState().state;
+  const { walletAccount } = useGlobalState().state;
+  const [loading, setLoading] = useState<boolean>(false);
+  const [amount, setAmount] = useState<string>("");
+  const [step, setStep] = useState<ClaimStep>(ClaimStep.Start);
+  const available = nativeToDecimal(userRewardsBalance);
+  const nextStep = step == ClaimStep.Success ? undefined : step + 1;
+
+  useMemo(() => {
+    if (!visible)
+      setTimeout(() => {
+        setStep(ClaimStep.Start)
+        setAmount("")
+      }, 500)
+  }, [visible])
+
+
+  const submitExtrinsic = useCallback(() => {
+    if (!walletAccount || !api || !amount)
+      return;
+
+    const extrinsic = createClaimRewardExtrinsic(amount);
+
+    extrinsic
+      ?.signAndSend(
+        walletAccount.address,
+        { signer: walletAccount.signer as any },
+        (result) => {
+          const { status, events } = result;
+
+          const errors = getErrors(events, api);
+          if (status.isInBlock) {
+            if (errors.length > 0) {
+              const errorMessage = `Transaction failed with errors: ${errors.join(
+                "\n"
+              )}`;
+              console.error(errorMessage);
+              toast(errorMessage, { type: "error" });
+            }
+          } else if (status.isFinalized) {
+            setLoading(true)
+
+            if (errors.length === 0) {
+              setStep(ClaimStep.Success)
+            }
+          }
+        }
+      )
+      .catch((error) => {
+        console.error("Transaction submission failed", error);
+        toast("Transaction submission failed", { type: "error" });
+        setLoading(false);
+      });
+  }, [api, amount, createClaimRewardExtrinsic,
+    walletAccount, setLoading, setStep]);
+
+
+  const getContentForStep = (step: ClaimStep) => {
```

Comment:
Maybe it's better if we memoize this as 
```
const Content = useMemo(() => {
switch step ...
, [step, ...])
```

Reply:
Uhm, I don't think it'll make a big difference, but okay.

Reply:
Okay I've changed it and now its memoized, I restored some logic that I have accidentally deleted too in this commit.